source/scripts/Background/controllers/message-handler/popup-promise.ts

```diff
@@ -17,14 +17,22 @@ export const popupPromise = async ({
   host,
   route,
 }: {
-  data?: object;
+  data?: object | any;
   eventName: string;
   host: string;
   route: string;
 }) => {
-  const { dapp, createPopup } = window.controller;
-  if (eventName !== 'connect' && !dapp.isConnected(host)) return;
-  if (dapp.hasWindow(host)) return;
+  const { dapp, createPopup, wallet } = window.controller;
+  const currentNetwork = wallet.getNetwork();
+  if (
+    (eventName === 'wallet_switchEthereumChain' &&
+      data.chainId === currentNetwork.chainId) ||
+    (dapp.hasWindow(host) && eventName === 'connect') ||
+    (!dapp.isConnected(host) && eventName !== 'connect') ||
+    (dapp.isConnected(host) && eventName === 'connect') ||
```

Comment:
This connect event check needs to be refactored as well it might lead to unexpected behaviors. If the Dapp isConnected and the wallet is locked the screen should open to allow the user to unlock the wallet, see the below sections of code: 
https://github.com/syscoin/pali-wallet/blob/54be72b301c0570347eecb3efe95b99bb3337783/source/scripts/Background/controllers/message-handler/requests.ts#L217

https://github.com/syscoin/pali-wallet/blob/54be72b301c0570347eecb3efe95b99bb3337783/source/pages/Connections/ConnectWallet.tsx#L42